pennylane/workflow/construct_execution_config.py

```diff
@@ -105,10 +105,7 @@ def wrapper(*args, **kwargs):
 
         if resolve:
             tape = construct_tape(qnode, level=0)(*args, **kwargs)
-            # pylint:disable=protected-access
-            config = _resolve_execution_config(
-                config, qnode.device, (tape,), qnode._transform_program
-            )
+            config = _resolve_execution_config(config, qnode.device, (tape,))
```

Comment:
Still weird that `construct_batch` causes isssues when `construct_tape` doesn't, but let's just do:

```suggestion
            tape = qml.tape.make_qscript(qnode.func)(*args, **kwargs)
            batch, _ = qnode.transform_program((tape,))
            config = _resolve_execution_config(config, qnode.device, batch)
```
To remove a dependency on both.

Reply:
Potentially with a `from pennylane.tape import make_qscript` at the top to be explicit with the import.